Formula/lib/libnghttp3.rb

```diff
@@ -0,0 +1,35 @@
+class Libnghttp3 < Formula
+  desc "HTTP/3 library written in C"
+  homepage "https://nghttp2.org/nghttp3/"
+  url "https://github.com/ngtcp2/nghttp3/archive/refs/tags/v0.15.0.tar.gz"
+  sha256 "6a75f5563e58d99e6b98d442111ac677984011c66b8b4f923764712741399027"
+  license "MIT"
+
+  head do
+    url "https://github.com/ngtcp2/nghttp3.git", branch: "main"
+  end
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, "-DENABLE_LIB_ONLY=1"
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <nghttp3/nghttp3.h>
+      #include <stdio.h>
+
+      int main(void) {
+        const nghttp3_info *info = nghttp3_version(0);
+        printf("%s", info->version_str);
+        return 0;
+      }
+    EOS
+
+    system ENV.cc, "test.c", "-I#{include}", "-L#{lib}", "-lnghttp3", "-o", "test"
```

Comment:
Does the install generate pkg-config files? If so, we can test those here by doing
```suggestion
    flags = shell_output("pkg-config --cflags --libs nghttp3").chomp.split
    system ENV.cc, "test.c", *flags, "-o", "test"
```

Reply:
It does! I'll include this suggestion, thanks! I've also reworked the tests locally to work with `--head`, will push that in a moment with this suggestion incorporated.